pkg/quarterdeck/apikeys.go

```diff
@@ -168,6 +170,16 @@ func (s *Server) APIKeyCreate(c *gin.Context) {
 		return
 	}
 
+	// Ensure the user cannot create api keys that have permissions they do not.
+	for _, permission := range key.Permissions {
+		if perms.UserKeyPermission(permission) && !claims.HasPermission(permission) {
+			// Do not allow the user to create an apikey with this permission
+			log.Warn().Msg("user attempted to create api key with permissions they didn't have")
+			c.JSON(http.StatusBadRequest, api.ErrorResponse("invalid permissions requested for apikey"))
+			return
+		}
+	}
```

Comment:
@pdeziel this should fix SC-14746. There is a related test below.

Reply:
So right now there is a test that uses the database to make sure the naive method is working; the naive method is more performant and if we add the check to Tenant that will just be more good reasons to keep the naive method backed by the test. I'd advocate for adding the check to tenant and making a note that nothing in the `permissions` package should use the database.